colcon_package_selection/package_discovery/ignore.py

```diff
@@ -51,6 +53,17 @@ def augment_packages(
                     'compile: {e}'.format_map(locals()))
                 self._args.packages_ignore_regex.remove(pattern)
 
+            if not any(re.match(pattern, pkg_name) for pkg_name in pkg_names):
+                logger.warn(
+                    "the --packages-ignore-regex '{pattern}' doesn't match "
+                    'any of the package names'.format_map(locals()))
+
+        for pkg_name in (self._args.packages_ignore or []):
+            if pkg_name not in pkg_names:
+                logger.warn(
+                    "ignoring unknown package '{pkg_name}' in "
```

Comment:
wording is weird IMO, some suggestions below though I'm not very satisfied with those either:
"requested to ignore package '{pkg_name}' but this package was not found"
"--packages-ignore '{pkg_name}' doesn't match any of the found packages"